### PR TITLE
MAINT: Modifies minute bars to use a dict of OHLC ratios

### PR DESCRIPTION
**Pull request by yankees714**
Instead of using a single ratio to scale up pricing for each sid before writing for bcolz, the writer now accepts a dict mapping each sid to the ratio to use. This arg is still optional, and we use the existing
`OHLC_RATIO` (1000) for every sid if no dict is passed.

This allows better handling of futures pricing data, where the required precision across root symbols is not consistent.

### CODE REVIEW

File: zipline/data/minute_bars.py [hunk 1]
```diff
@@ -265,7 +274,8 @@ def read(cls, rootdir):
 
     def __init__(
         self,
-        ohlc_ratio,
+        default_ohlc_ratio,
```
Comment (ehebert) [inline zipline/data/minute_bars.py]: Not 100% sure we need to put the prefix of `default` here, but not going to block on it.

File: zipline/data/minute_bars.py [hunk 2]
```diff
@@ -465,6 +489,17 @@ def __init__(self,
     def first_trading_day(self):
         return self._start_session
 
+    def ohlc_ratio_for_sid(self, sid):
+        if self._ohlc_ratios_per_sid is not None:
+            try:
+                return self._ohlc_ratios_per_sid[sid]
+            except KeyError:
```
Comment (ehebert) [inline zipline/data/minute_bars.py]: If the `_default_ohlc_ratio` is being used as a fallback/default, should it be returned here?

File: zipline/data/minute_bars.py [hunk 3]
```diff
@@ -856,6 +897,17 @@ def last_available_dt(self):
     def first_trading_day(self):
         return self._start_session
 
+    def _ohlcv_ratio_inverse_for_sid(self, sid):
+        if self._ohlc_inverses_per_sid is not None:
+            try:
+                return self._ohlc_inverses_per_sid[sid]
+            except KeyError:
+                pass
```
Comment (ehebert) [inline zipline/data/minute_bars.py]: Same as with `ohclv_ratio_for_sid`, should the `_default_ohlc_inverse` be returned here?

File: zipline/data/minute_bars.py (hunk 2 above)
Comment (yankees714) [inline zipline/data/minute_bars.py]: I opted here to share the return below for both the case where the sid is not present in the dict, and when no dict was passed to begin with. Do you think separate returns would be better?

File: zipline/data/minute_bars.py (hunk 1 above)
Comment (yankees714) [inline zipline/data/minute_bars.py]: Was trying to make the distinction from `ohlc_ratios_per_sid` clear in the naming, but if it's overkill I'd be happy to simplify it.

File: zipline/data/minute_bars.py [hunk 4]
```diff
@@ -1111,11 +1164,15 @@ def load_raw_arrays(self, fields, start_dt, end_dt, sids):
...
                             excl_start - start_idx:excl_stop - start_idx + 1]
                         values = np.delete(values, excl_slice)
+
                 where = values != 0
                 # first slice down to len(where) because we might not have
                 # written data for all the minutes requested
-                out[:len(where), i][where] = values[where]
-            if field != 'volume':
-                out *= self._ohlc_inverse
+                if field != 'volume':
```
Comment (ehebert) [inline zipline/data/minute_bars.py]: Why the change on how the data is written to the index?
I lean towards the original, where the assignment is done in one place, to minimize drift between OHLC and Volume with regard to the write indices.

Comment (ehebert) [inline zipline/data/minute_bars.py]: I understand now, since the ratio is per sid, the assignment had to be changed to within the loop.

File: zipline/data/minute_bars.py (hunk 2 above)
Comment (ehebert) [inline zipline/data/minute_bars.py]: Never mind my comment, the `pass` does fall through correctly. It did read slightly off to me at first, but I believe passing on the KeyError is the correct way to go about this.
